reference/dic_tags.rst

```diff
@@ -227,6 +228,107 @@ assetic.templating.twig
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="app.mysql_lock" class="AppBundle\Lock\MysqlLock" />
+                <service id="app.postgresql_lock" class="AppBundle\Lock\PostgresqlLock" />
+                <service id="app.sqlite_lock" class="AppBundle\Lock\SqliteLock" />
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        $container
+            ->register('app.mysql_lock', 'AppBundle\Lock\MysqlLock')
+            ->register('app.postgresql_lock', 'AppBundle\Lock\PostgresqlLock')
+            ->register('app.sqlite_lock', 'AppBundle\Lock\SqliteLock')
+        ;
+
+Instead of dealing with these three services, your application needs a generic
+``app.lock`` service. This service must be an alias to any of the other services.
+Thanks to the ``auto_alias`` option, you can automatically create that alias
+based on the value of a configuration parameter.
+
+Considering that a configuration parameter called ``database_type`` exists,
+the generic ``app.lock`` service can be defined as follows:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        services:
+            app.mysql_lock:
+                class: AppBundle\Lock\MysqlLock
+            app.postgresql_lock:
+                class: AppBundle\Lock\PostgresqlLock
+            app.sqlite_lock:
+                class: AppBundle\Lock\SqliteLock
+            app.lock:
+                tags:
+                    - { name: auto_alias, format: "%database_type%.lock" }
+
+    .. code-block:: xml
+
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="app.mysql_lock" class="AppBundle\Lock\MysqlLock" />
+                <service id="app.postgresql_lock" class="AppBundle\Lock\PostgresqlLock" />
+                <service id="app.sqlite_lock" class="AppBundle\Lock\SqliteLock" />
+
+                <service id="app.lock">
+                    <tag name="auto_alias" format="%database_type%.lock" />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        $container
+            ->register('app.mysql_lock', 'AppBundle\Lock\MysqlLock')
+            ->register('app.postgresql_lock', 'AppBundle\Lock\PostgresqlLock')
+            ->register('app.sqlite_lock', 'AppBundle\Lock\SqliteLock')
+
+            ->register('app.lock')
+            ->addTag('auto_alias', array('format' => '%database_type%.lock'))
```

Comment:
the format is wrong here according to your lock implementation service ids. It should be ``app.%database_type%_lock``